src/firestarter/Environment/X86/Payload/X86Payload.cpp

```diff
@@ -89,3 +90,397 @@ X86Payload::highLoadFunction(unsigned long long *addrMem,
+                    std::is_same<asmjit::x86::Gpq, IterReg>::value,
+                "IterReg is not of any supported type");
+
+  if constexpr (std::is_same<asmjit::x86::Mm, IterReg>::value) {
+    assert((iter_reg == asmjit::x86::mm0, "iter_reg must be mm0"));
+  }
+
+  assert((iter_reg != temp_reg, "iter_reg must be != temp_reg"));
+  assert((temp_reg != temp_reg2, "temp_reg must be != temp_reg2"));
+  assert((temp_reg != addrHigh_reg, "temp_reg must be != addrHigh_reg"));
+  assert((temp_reg != pointer_reg, "temp_reg must be != pointer_reg"));
+
+  assert((iter_reg != asmjit::x86::r8, "iter_reg must be != r8"));
+  assert((iter_reg != asmjit::x86::r9, "iter_reg must be != r9"));
+  assert((iter_reg != asmjit::x86::rax, "iter_reg must be != rax"));
+  assert((iter_reg != asmjit::x86::rbx, "iter_reg must be != rbx"));
+  assert((iter_reg != asmjit::x86::rcx, "iter_reg must be != rcx"));
+  assert((iter_reg != asmjit::x86::rdx, "iter_reg must be != rdx"));
+
+  assert((temp_reg != asmjit::x86::r8, "temp_reg must be != r8"));
+  assert((temp_reg != asmjit::x86::r9, "temp_reg must be != r9"));
+  assert((temp_reg != asmjit::x86::rax, "temp_reg must be != rax"));
+  assert((temp_reg != asmjit::x86::rbx, "temp_reg must be != rbx"));
+  assert((temp_reg != asmjit::x86::rcx, "temp_reg must be != rcx"));
+  assert((temp_reg != asmjit::x86::rdx, "temp_reg must be != rdx"));
+
+  assert((temp_reg2 != asmjit::x86::r8, "temp_reg2 must be != r8"));
+  assert((temp_reg2 != asmjit::x86::r9, "temp_reg2 must be != r9"));
+  assert((temp_reg2 != asmjit::x86::rax, "temp_reg2 must be != rax"));
+  assert((temp_reg2 != asmjit::x86::rbx, "temp_reg2 must be != rbx"));
+  assert((temp_reg2 != asmjit::x86::rcx, "temp_reg2 must be != rcx"));
+  assert((temp_reg2 != asmjit::x86::rdx, "temp_reg2 must be != rdx"));
+
+  assert((addrHigh_reg != asmjit::x86::r8, "addrHigh_reg must be != r8"));
+  assert((addrHigh_reg != asmjit::x86::r9, "addrHigh_reg must be != r9"));
+  assert((addrHigh_reg != asmjit::x86::rax, "addrHigh_reg must be != rax"));
+  assert((addrHigh_reg != asmjit::x86::rbx, "addrHigh_reg must be != rbx"));
+  assert((addrHigh_reg != asmjit::x86::rcx, "addrHigh_reg must be != rcx"));
+  assert((addrHigh_reg != asmjit::x86::rdx, "addrHigh_reg must be != rdx"));
+
+  auto SkipErrorDetection = cb.newLabel();
+
+  if constexpr (std::is_same<asmjit::x86::Mm, IterReg>::value) {
+    cb.movq(temp_reg, iter_reg);
+  } else {
+    cb.mov(temp_reg, iter_reg);
+  }
+  // round about 50-100 Hz
+  // more or less, but this isn't really that relevant
+  cb.and_(temp_reg, asmjit::Imm(0x3fff));
+  cb.test(temp_reg, temp_reg);
+  cb.jnz(SkipErrorDetection);
+
+  cb.mov(temp_reg, asmjit::Imm(0xffffffff));
+
+  int registerCount = (int)this->registerCount();
+
+  // Create a backup of VectorReg(0)
+  if constexpr (std::is_same<asmjit::x86::Xmm, VectorReg>::value) {
+    cb.movq(temp_reg2, asmjit::x86::xmm0);
+    cb.push(temp_reg2);
+    cb.crc32(temp_reg, temp_reg2);
```

Comment:
please do not use crc32, since it is not part of the x86_64 ISA

Reply:
A check for SSE4.2 was added. It is available on Intel since Nehalem and AMD since Bulldozer.